Create ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md (#2078)
SoftwareAG/c8y-docs: content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md

---
date: ""

> title: Fix type error when using `hookDeviceListColumns` with factory

Author: The Claude AI suggested to use the following title:
```suggestion
title: Fixed type error when using hookDeviceListColumns with factory
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-3308
version: 1019.24.6
---

> Fix type error when using `hookDeviceListColumns` with factory

Author: The Claude AI suggested to use the following body:
```suggestion
In the Cumulocity IoT platform, the hookDeviceListColumns API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, preventing the customization from working as expected. This issue has now been resolved. With this fix, users can successfully utilize the hookDeviceListColumns API in combination with a factory to customize the device list columns according to their requirements without encountering any type errors.
```

> Fix type error when using `hookDeviceListColumns` with factory

Reviewer: ```suggestion
In the Cumulocity IoT platform, the hookDeviceListColumns API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.
```

> In Cumulocity IoT, the ‘hookDeviceListColumns‘ API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.

Reviewer 2: ```suggestion
In Cumulocity IoT, the `hookDeviceListColumns` API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.
```
